### ISSUE TITLE
URL class option 'alphabetize GET parameters' is no longer respected

### ISSUE THREAD

**Issue by GunnerGuyven** [hydrusnetwork/hydrus#1551]
### Hydrus version

574

### Qt major version

Qt 6

### Operating system

Linux (specify distro and version in comments)

### Install method

Third party (AUR, Docker, Chocolatey, etc. Specify in comments)

### Install and OS comments

Arch AUR

### Bug description and reproduction

The setting in question is found at:
network > downloader components > manage URL classes > edit URL class > options > alphabetize GET parameters when normalizing

This option worked until recently (I am not sure which version the regression happened in).  This option appears to be no longer respected, resulting in failures when accessing particularly strict servers.

Option Off:
![image](https://github.com/hydrusnetwork/hydrus/assets/543524/58bdc554-2479-44ed-8fb3-1039891b455a)

Option On (for comparison, behavior is on always):
![image](https://github.com/hydrusnetwork/hydrus/assets/543524/b2eebff0-fa83-4ae0-b826-0de3605fd796)

**Comment by GeorgeEBeresford** [comment]
Did some experimenting and the issue was first present in the 570 release. The parameters aren't alphabetised in 569 with this option toggled off

**Comment by GeorgeEBeresford** [comment]
Sorry. Just checked and importing with 569 causes it to fail as the additional url is alphabetised (even though under url classes, it shows it as non-alphabetised).

It imports with the correct non-alphabetised field in 568 Test Build 3. Soon as it hits 569 Future Build, it alphabetises the parameters while importing (with the opton unchecked).

Hope that helps narrow the issue down!

**Comment by hydrusnetwork** [comment]
Thank you for this report. I am sorry for the trouble--when I recently updated the encoded URL handling, my master 'let's wash this URL as soon as it is ingested' method, which converts space to %20 and so on, was by default alphabetising all query parameters. This wasn't caught in URL unit tests because they couldn't see the forest for the trees. I have fixed it and added a new test.

Let me know if you have any more trouble with it in v575.
